static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,270 @@
+    return (
+      <div className="disaster-dashboard-map-section">
+        <div className="error-message">
+          Sorry, we do not have maps for this place.
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="disaster-dashboard-map-section">
+      <div className="zoom-button-section">
+        <div id={ZOOM_IN_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">add</i>
+        </div>
+        <div id={ZOOM_OUT_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">remove</i>
+        </div>
+      </div>
+      <div id={MAP_ID} ref={svgContainer}></div>
+      {props.selectedDisaster === DisasterType.ALL ? (
+        <div className="all-disasters-legend">
+          {Object.keys(DISASTER_COLORS).map((disasterType) => {
+            return (
+              <div className="all-disasters-legend-entry" key={disasterType}>
+                <div
+                  className="all-disasters-legend-color"
+                  style={{ backgroundColor: DISASTER_COLORS[disasterType] }}
+                ></div>
+                <span>{disasterType}</span>
+              </div>
+            );
+          })}
+        </div>
+      ) : (
+        <div id={LEGEND_CONTAINER_ID}></div>
+      )}
+      <div id="disaster-dashboard-info-card" ref={infoCardRef} />
+    </div>
+  );
+
+  function onPointClicked(
+    point: DisasterEventPoint,
+    event: PointerEvent
+  ): void {
+    const infoCard = d3.select(infoCardRef.current);
+    ReactDOM.render(
+      <InfoCard
+        onClose={() =>
+          d3.select(infoCardRef.current).style("visibility", "hidden")
+        }
+        eventData={point}
+      />,
+      infoCardRef.current
+    );
+    // get info card dimensions
+    const infoCardRect = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect();
+    const infoCardHeight = infoCardRect.height;
+    const infoCardWidth = infoCardRect.width;
+    const containerWidth = (
+      d3.select(svgContainer.current).node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    // calculate left and top position to show the info card
+    let left = Math.min(event.offsetX, containerWidth - infoCardWidth);
+    if (left < 0) {
+      left = 0;
+      infoCard.style("width", containerWidth + "px");
+    } else {
+      infoCard.style("width", "fit-content");
+    }
+    let top = event.offsetY - infoCardHeight - iNFO_CARD_OFFSET;
+    if (top < 0) {
+      top = event.offsetY + iNFO_CARD_OFFSET;
+    }
+    // set info card position and make it visible
+    infoCard
+      .style("left", left + "px")
+      .style("top", top + "px")
+      .style("visibility", "visible");
+  }
+
+  function canClickRegion(placeDcid: string): boolean {
+    // If a country is a European country, we want to add Europe to its list
+    // of parents. This is important for getAllChildPlaceTypes (line 191)
+    // because it will handle European countries differently than regular
```

Comment:
thanks for adding the comment.

nit: getParentPlaces is where it's used. referring to line numbers is brittles since that will change frequently :)

Reply:
done